src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -190,7 +190,8 @@ parseTypeClassDeclaration = do
 
 parseConstraint :: TokenParser Constraint
 parseConstraint = Constraint <$> parseQualified properName
-                             <*> P.many (noWildcards parseTypeAtom)
+                             <*> (P.many (noWildcards (parens parsePolyType)
```

Comment:
There's definitely an issue, I just think this isn't the right fix. The issue seems to be that it's parsed as 

```
Fail ((TypeConcat (TypeString a) "bla") => Show (MyType a))
```

but constrained types shouldn't be allowed in instance heads.